File: gateway-server/src/main/java/org/lucycato/gatewayserver/application/port/in/LoggingGlobalFilterUseCase.java
package org.lucycato.gatewayserver.application.port.in;

import org.lucycato.gatewayserver.application.port.in.command.LogRequestCommand;
import org.lucycato.gatewayserver.application.port.in.command.LogResponseCommand;
import reactor.core.publisher.Mono;

public interface LoggingGlobalFilterUseCase {
    Mono<Void> logRequest(LogRequestCommand command);

    Mono<Void> logResponse(LogResponseCommand command);
}

File: gateway-server/src/main/kotlin/org/lucycato/gatewayservice/application/port/in/LoggingGlobalFilterUseCase.kt
package org.lucycato.gatewayservice.application.port.`in`

import org.lucycato.gatewayservice.application.port.`in`.commnad.LogRequestCommand
import org.lucycato.gatewayservice.application.port.`in`.commnad.LogResponseCommand
import reactor.core.publisher.Mono

interface LoggingGlobalFilterUseCase {
    fun logRequest(command: LogRequestCommand): Mono<Void>

    fun logResponse(command: LogResponseCommand): Mono<Void>
}